### PR TITLE
[Scala3] Fix go to defition for givens

### PR DESCRIPTION
**Pull request by dos65**
In semanticdb given is a term if there is no type/using pameters,
otherwise it's a method.
Also, fixes name inference for anonymous givens.

### CODE REVIEW

File: tests/unit/src/test/resources/definition-scala3/example/GivenAlias.scala [hunk 1]
```diff
@@ -2,12 +2,49 @@ package example
...
+  def doY/*GivenAlias.scala*/: String/*Predef.scala*/
+
+trait Zg/*GivenAlias.scala*/[T/*GivenAlias.scala*/]:
+  def doZ/*GivenAlias.scala*/: List/*package.scala*/[T/*GivenAlias.scala*/]
+
+given Xg/*GivenAlias.scala*/ with
+  def doX/*GivenAlias.scala*/ = 7
+
+given (using/*<no symbol>*/ Xg/*GivenAlias.scala*/): Yg/*GivenAlias.scala*/ with
+  def doY/*<no file>*/ = "7"
```
Comment (tgodzik) [inline tests/unit/src/test/resources/definition-scala3/example/GivenAlias.scala:33]: Are w enot indexing the body of given here?
```suggestion
  def doY/*GivenAlias.scala*/ = "7"
```

File: tests/unit/src/test/resources/definition-scala3/example/GivenAlias.scala [hunk 2]
```diff
@@ -2,12 +2,49 @@ package example
...
+
+given [T/*<no file>*/]: Zg/*GivenAlias.scala*/[T/*<no file>*/] with
```
Comment (tgodzik) [inline tests/unit/src/test/resources/definition-scala3/example/GivenAlias.scala:35]: This shoul be fixed by the current PR, no ?
```suggestion
given [T/*GivenAlias.scala*/]: Zg/*GivenAlias.scala*/[T/*<no file>*/] with
```

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks for the fix! It seems that a couple of new problems popped up here, or is it expected?

File: tests/unit/src/test/resources/definition-scala3/example/GivenAlias.scala [hunk 3]
```diff
@@ -2,12 +2,49 @@ package example
 
 given intValue/*GivenAlias.scala*/: Int/*Int.scala*/ = 4
 given String/*Predef.scala*/ = "str"
+given (using/*<no symbol>*/ Int/*Int.scala*/): Double/*Double.scala*/ = 4.0
+given [T/*GivenAlias.scala*/]: List/*package.scala*/[T/*GivenAlias.scala*/] = Nil/*package.scala*/
+given given_Char/*GivenAlias.scala*/: Char/*Char.scala*/ = '?'
+given `given_Float`/*<no symbol>*/: Float/*Float.scala*/ = 3.0
```
[the remark this answers is not in the document]
Comment (dos65) [reply tests/unit/src/test/resources/definition-scala3/example/GivenAlias.scala:8]: There is a difference with Scala2 in positions for backquoted ident.
In Scala3 range refers only to name, while in Scala2 it includes backquotes.

I've raised an issue - https://github.com/lampepfl/dotty/issues/14012

**Review by tgodzik — APPROVED** [review]
Looks awesome! I just added one question, but otherwise LGTM

File: mtags/src/main/scala/scala/meta/internal/mtags/ScalaMtags.scala [hunk 4]
```diff
@@ -120,6 +120,39 @@ class ScalaMtags(val input: Input.VirtualFile, dialect: Dialect)
...
+            if (tparams.nonEmpty || paramss.nonEmpty) {
+              method(name, "()", pos, Property.IMPLICIT.value)
+              enterTypeParameters(tparams)
+              enterTermParameters(paramss, isPrimaryCtor = false)
+
+            } else {
+              term(name, pos, Kind.METHOD, Property.IMPLICIT.value)
+            }
+          }
+          if (!isAlias) {
```
[the remark this answers is not in the document]
Comment (dos65) [reply mtags/src/main/scala/scala/meta/internal/mtags/ScalaMtags.scala]: Thx! I split it onto two different methods. It wasn't fully correct as some non-existing symbols were introduced

File: mtags/src/main/scala/scala/meta/internal/mtags/ScalaMtags.scala [hunk 5]
```diff
@@ -303,13 +347,9 @@ class ScalaMtags(val input: Input.VirtualFile, dialect: Dialect)
                 Some((t.name.value, t.name.pos))
             }
 
+          owner
```
Comment (tgodzik) [inline mtags/src/main/scala/scala/meta/internal/mtags/ScalaMtags.scala:350]:
```suggestion
```

this looks unused?

Comment (dos65) [reply mtags/src/main/scala/scala/meta/internal/mtags/ScalaMtags.scala:350]: oops! thx!

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**